### ISSUE TITLE
Forward Errors in Paper

### ISSUE THREAD

**Issue by MamonaAwan** [ESanchezLozano/SAIC-Unsupervised-landmark-detection-NeurIPS2019#2]
Hi, I tried to reproduce the forward errors that are mentioned in the paper. The output that I get for last epoch is as below:
{'79': [array([[0.08620551, 0.10066307, 0.07971909, 0.07951361, 0.09236273,
        0.10396701, 0.089784  , 0.08467723, 0.0842989 , 0.10591685],
       [0.13973934, 0.13489833, 0.08176813, 0.09055925, 0.09529157,
        0.10192109, 0.14680529, 0.08288474, 0.09594048, 0.11188923],
       [0.1300454 , 0.08081679, 0.11438627, 0.14266052, 0.09695703,
        0.13732754, 0.0877717 , 0.1014945 , 0.10404468, 0.13985691],
       [0.06358634, 0.0696696 , 0.06553751, 0.06716975, 0.06696829,
        0.06533571, 0.06474016, 0.06772308, 0.06787112, 0.07103965],
       [0.0595727 , 0.0587818 , 0.05895047, 0.05975251, 0.05957813,
        0.0586871 , 0.05958021, 0.05910266, 0.059256  , 0.05908673],
       [0.05907548, 0.0586444 , 0.05857861, 0.05835644, 0.05836652,
        0.05796919, 0.0584557 , 0.0592101 , 0.05847388, 0.0582365 ],
       [0.05790327, 0.05792443, 0.05774715, 0.05777053, 0.05784382,
        0.05769782, 0.05784153, 0.05795715, 0.05789781, 0.05778359],
       [0.05767939, 0.05767939, 0.0576794 , 0.05767945, 0.05767941,
        0.05767939, 0.05767941, 0.05767941, 0.05767939, 0.05767941]])]}

However, the errors reported in the paper are in different configuration. Can you please enlighten how do you calculate the errors in table 1 & 2 from these errors.

Thanks a lot,

**Comment by ESanchezLozano** [comment]
This is because the script returns the results for each of the repetitions on the forward experiment (you have to average them). When few images are used to train the forward regressor the variance of the results is going to be larger. When using all the images to train the regressor as you can see in the last row, the variance is just due to randomness in shuffling the data (line 53 in test.py), and it is very small. 

I hope this helps. If not, please feel free to let me know :)

**Comment by MamonaAwan** [comment]
Thank you for your prompt response, I do get that rows are for number of images used, and columns are for number of points. If I average them, for instance average of last row (all images) is 0.0576, But if you see table 2, its reported as 3.99. (table 2, row ALL, MAFL, Proposed).

**Comment by ESanchezLozano** [comment]
Is it possible that your trained model contains some unstable points? In some cases when this is the case the errors can appear larger. In my experience using the approach we propose in the paper it is less likely that outliers will appear than when trained from scratch or finetuning. You can see this measuring the affine errors.

**Comment by MamonaAwan** [comment]
I trained this on CelebA and followed the instructions as mentioned since there wasn't any pretrained model. May be that's the case, so in this case its 5.76?

**Comment by ESanchezLozano** [comment]
The pretrained model is the one that is attached to the github code. I guess that the model you trained is the default with incremental learning. If this is the case, may I suggest you to train several models (training is very fast in the incremental model), and check on their variation on the test measures?

**Comment by MamonaAwan** [comment]
Sorry for being vague, The pretrained model that's attached is the 'core' model. I was referring to Pretrained model of  'Proposed' method. I have actually trained several models and that's why I was confused about its correspondence in paper. I'll give it another try. Thanks a lot for your time.
